setup.py

```diff
@@ -53,7 +53,7 @@
         'Topic :: Software Development :: Libraries :: Python Modules'
     ],
     install_requires=[
-        'Django >=2.1,<5.2',
+        'Django >=2.1,<6.0',
```

Comment:
IMO the requirements here should be bumped to `>=4.2` since this is now the minimum officially supported version of Django and also the minimum version we test with. @tomturner WDYT ?

Reply:
@atodorov thats fine I agree.  If you don't mind looking at the tests that would be great.  I am quite busy at the moment and I will be watching the Django con talks this week.

I did a seperate branch nothing different to this in the end and I found that i needed the `autodetector = MigrationAutodetector` line.

Reply:
> @atodorov thats fine I agree. If you don't mind looking at the tests that would be great.

FTR I took a quick look b/c I have lots of failing items on my side. It appears that `_fixture_setup()` has been changed from an instance method to a class method + there are additional changes which I still don't quite understand. 

Relevant upstream ticket: https://code.djangoproject.com/ticket/35660

It's true we've been depending on an internal method for some customized testing needs and this is now broken. I will circle back to this once I have more understanding of the new behavior.

Reply:
Somehow the app registration is meshed up, adding the registration to `_pre_setup` solves it but I don't know why. However, we can no longer rely on _pre_setup because its signature has changed, making it incompatible. Therefore, we need to either implement some Django-specific version logic or find an alternative setup method.

```python
class TenantSyncTest(BaseSyncTest):
    @classmethod
    def _pre_setup(cls):
        cls.sync_shared()
        cls.INSTALLED_APPS = cls.SHARED_APPS + cls.TENANT_APPS

        settings.SHARED_APPS = cls.SHARED_APPS
        settings.TENANT_APPS = cls.TENANT_APPS
        settings.INSTALLED_APPS = cls.INSTALLED_APPS

        cls.available_apps = cls.INSTALLED_APPS

        super()._pre_setup()
```

Reply:
Found a solution: https://github.com/django-tenants/django-tenants/pull/1148